sentry/src/main/java/io/sentry/DuplicateEventDetectionEventProcessor.java

```diff
@@ -2,45 +2,63 @@
 
 import io.sentry.util.Objects;
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
-import java.util.Map;
-import java.util.WeakHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
 
 /** Deduplicates events containing throwable that has been already processed. */
 public final class DuplicateEventDetectionEventProcessor implements EventProcessor {
-  private final WeakHashMap<Throwable, Object> capturedObjects = new WeakHashMap<>();
+  private final ConcurrentLinkedQueue<Throwable> capturedObjects = new ConcurrentLinkedQueue<>();
   private final SentryOptions options;
+  private final int bufferSize;
 
   public DuplicateEventDetectionEventProcessor(final @NotNull SentryOptions options) {
+    this(options, 100);
+  }
+
+  public DuplicateEventDetectionEventProcessor(
+      final @NotNull SentryOptions options, int bufferSize) {
     this.options = Objects.requireNonNull(options, "options are required");
+    this.bufferSize = bufferSize;
   }
 
   @Override
   public SentryEvent process(final @NotNull SentryEvent event, final @Nullable Object hint) {
-    final Throwable throwable = event.getOriginThrowable();
-    if (throwable != null) {
-      if (capturedObjects.containsKey(throwable)
-          || containsAnyKey(capturedObjects, allCauses(throwable))) {
-        options
-            .getLogger()
-            .log(
-                SentryLevel.DEBUG,
-                "Duplicate Exception detected. Event %s will be discarded.",
-                event.getEventId());
-        return null;
-      } else {
-        capturedObjects.put(throwable, null);
+    if (options.isEnableDeduplication()) {
```

Comment:
maybe add an `else` block and log it if it's disabled, we do this for all other integrations